.github/workflows/pull_request_build.yml

```diff
@@ -29,10 +29,12 @@ jobs:
         id: cache-data
         uses: actions/cache@v2
         with:
-          path: data/cats_dogs
-          key: ${{ hashFiles('data/cats_dogs.dvc') }}
+          path: .dvc/cache
+          key: ${{ runner.os }}-data-${{ hashFiles('**/*.dvc') }}
       - name: download data
         if: steps.cache-data.outputs.cache-hit != 'true'
+        run: dvc fetch data/cats_dogs.dvc
+      - name: checkout data
```

Comment:
Name is not adequate to what this step is doing.
Also, now I wonder whether we need fetch step at all? If we restored the cache, what `pull` should be doing is only repopulating state, right? Maybe we just need to keep the `pull` step?